Add Intruder game mode

FlameArrow57 opened the pull request:
[GAME MODES][FEATURE]

## About the PR <!-- Describe the Pull Request here. What does it change? What other things could this impact? -->
Adds the Intruder game mode per below
Design doc - https://hackmd.io/@FlameArrow57/ryuQFOQ-T
Player Project - https://discord.com/channels/182249960895545344/1358207672914280563


## Why's this needed? <!-- Describe why you think this should be added to the game. -->
Would be cool

![intruder](https://github.com/user-attachments/assets/eaa139f2-90c0-4ca2-bd47-e760c0ad6526)

## Changelog <!-- If necessary, put your changelog entry below. Otherwise, /please/ delete this entire section. -->
<!-- Put how you want to be credited in the changelog in place of CodeDude. -->
<!-- Use (*) for major changes and (+) for minor changes. See the contributor guide for details. For example: -->

```changelog
(u)FlameArrow57
(*)TBD
```

TobleroneSwordfish commented on code/datums/abilities/critter/mindeater.dm:
```diff
@@ -0,0 +1,390 @@
+
+	cast(atom/target)
+		. = ..()
+		actions.start(new /datum/action/bar/private/mindeater_brain_drain(target), src.holder.owner)
```
This doesn't require line of sight to be maintained and gives no feedback to the victim, meaning it seems to be possible to 100-0 brain drain someone while fully invisible and hiding behind a wall.

FlameArrow57 replied:
I was planning to add more feedback in the next stage of mindeaters if all goes well with the initial playtest, I'll add some more "temporary" feedback rn

TobleroneSwordfish commented on code/modules/projectiles/special.dm:
```diff
@@ -1400,3 +1400,42 @@ ABSTRACT_TYPE(/datum/projectile/special)
+			return
+		L.changeStatus("staggered", 0.75 SECONDS)
+		L.setStatus("psi_bolt_slow", 5 SECONDS)
+		var/datum/reagents/reagents = L.reagents
+		if (reagents)
+			var/amt = min(L.reagents.total_volume, 5)
+			L.reagents.remove_any_except(amt, "toxin")
+			L.reagents.add_reagent("toxin", amt)
```
This doesn't seem to be working?

FlameArrow57 replied:
Bug on master, being fixed in #22956

FlameArrow57 replied:
Okay fixed now

TobleroneSwordfish commented on code/modules/projectiles/special.dm:
```diff
@@ -1400,3 +1400,42 @@ ABSTRACT_TYPE(/datum/projectile/special)
+		if (istype(L, /mob/living/critter/mindeater))
+			return
+		L.changeStatus("staggered", 0.75 SECONDS)
+		L.setStatus("psi_bolt_slow", 5 SECONDS)
```
This slow is almost unnoticeable, especially with sprint.

FlameArrow57 replied:
Was bugged, LMK if you think it needs to be increased

TobleroneSwordfish commented on code/datums/abilities/critter/mindeater.dm:
```diff
@@ -0,0 +1,390 @@
+
+/datum/targetable/critter/mindeater/confuse
+
+/datum/targetable/critter/mindeater/shades
```
This is a really cool ability but movement (especially diagonal?) seems to immediately reveal which is the real one because the others don't animate properly.

FlameArrow57 replied:
Any suggestions on how to improve? I tried a few things and manually calling Move() seemed to be the best. Could maybe replace /obj/dummy/fake_mindeater with a critter version and have movement_delay() be set to the parent mindeater's movement delay, but maybe better post-initial-playtest?

TobleroneSwordfish commented on code/datums/abilities/critter/mindeater.dm:
```diff
@@ -0,0 +1,390 @@
+	cast(atom/target)
+		. = ..()
+		var/mob/living/critter/mindeater/mindeater = src.holder.owner
+		mindeater.undisguise()
```
Being undisguised by shooting doesn't revert this ability.

TobleroneSwordfish reviewed (changes requested):
- Pierce the veil goes on cooldown when interrupted or used on a corpse, probably not intentional?
- Disguise still uses the wrong ability icon when cancelled
- Disguise instantiates an entire new human for every disguise and can be spammed, since humans tend not to GC this is a potentially server crashing memory leak if someone autoclickers it. Consider keeping one dummy human and re-rolling their appearance repeatedly.
- Disguise shows your voice as `mindeater (as Unknown)`, probably should use your disguised name?
- Paralyze goes on full cooldown when the target is looking away, potentially quite frustrating? I have my doubts about using facing direction in SS13 in general but it might be neat to try out.
- Pierce the veil needs to hold a reference to the allocated region (probably on the status effect?) or it will be marked to be cleared and overwritten by other veil regions or misc allocated zones.

FlameArrow57 commented:
> * Pierce the veil goes on cooldown when interrupted or used on a corpse, probably not intentional?
Can you clarify when interrupted? It's meant to go on cooldown if using another ability during it or if your 3 shield charges run out. Fixed using on a corpse

> * Disguise still uses the wrong ability icon when cancelled
Fixed

> * Disguise instantiates an entire new human for every disguise and can be spammed, since humans tend not to GC this is a potentially server crashing memory leak if someone autoclickers it. Consider keeping one dummy human and re-rolling their appearance repeatedly.
Done

> * Disguise shows your voice as `mindeater (as Unknown)`, probably should use your disguised name?
> * Paralyze goes on full cooldown when the target is looking away, potentially quite frustrating? I have my doubts about using facing direction in SS13 in general but it might be neat to try out.
Maybe, it's meant to be intentional. People can change their direction to try to bait it out since it is more of a powerful ability

> * Pierce the veil needs to hold a reference to the allocated region (probably on the status effect?) or it will be marked to be cleared and overwritten by other veil regions or misc allocated zones.
I think it is referenced? On the "mindeater_abducted" status effect, var "alloc_region"
